internal/pretty/pretty.go

```diff
@@ -36,17 +35,17 @@ const jsonIndent = "  "
 // If marshal fails, it falls back to fmt.Sprintf("%+v").
 func ToJSON(e any) string {
 	switch ee := e.(type) {
-	case protov1.Message:
+	case protoadapt.MessageV1:
```

Comment:
nit: can we combine some of this now from below `case: protoadapt.MessageV2`?

Reply:
Do you mean removing the switch and receive a `proto.message` as ToJSON argument?

Reply:
+1; it would be nice to unify these two codepaths completely, but feel free to disregard if you don't want to do it.

I.e.

```go
if ee, ok := e.(protoadapt.MessageV1); ok {
	e = protoadapt.MessageV2Of(e)
}
switch ee := e.(type) {
	// no case for MessageV1 now
...
```

I don't know why the existing code sets different `MarshalOptions` but I highly suspect it's an oversight caused by having two codepaths that do essentially the same thing.